### PR TITLE
Updates the Fock backend ops.py to use the new Walrus gate signatures

### PR DESCRIPTION
**Pull request by nquesada**
The signature and return values of the `fock_gradients` functions is being modified in this PR https://github.com/XanaduAI/thewalrus/pull/164 . Once PR 164 of The Walrus is merged and a new release is made we will need to update the way these functions are used in SF. This PR does that.

### CODE REVIEW

File: strawberryfields/backends/fockbackend/ops.py [hunk 1]
```diff
@@ -326,7 +331,7 @@ def phase(theta, trunc):
     """
     return np.array(np.diag([exp(1j * n * theta) for n in range(trunc)]), dtype=def_type)
```
Comment (thisac) [inline strawberryfields/backends/fockbackend/ops.py:333]:
```suggestion


```

File: strawberryfields/backends/fockbackend/ops.py [hunk 2]
```diff
@@ -327,6 +332,7 @@ def phase(theta, trunc):
...
```
Comment (thisac) [inline strawberryfields/backends/fockbackend/ops.py:334]:
```suggestion


```

File: doc/gallery/minimizing_correlations/minimizing_correlations.ipynb [hunk 3]
```diff
@@ -689,9 +689,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python (tensorflow1.3)",
+   "display_name": "Python 3",
    "language": "python",
-   "name": "tensorflow1.3"
+   "name": "python3"
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/gallery/minimizing_correlations/minimizing_correlations.ipynb]: This PR will be on hold until the next walrus release is made, so plenty of time to fix up the merge conflicts :)

File: strawberryfields/backends/fockbackend/ops.py [hunk 4]
```diff
@@ -338,9 +344,9 @@ def beamsplitter(t, r, phi, trunc):
     # pylint: disable=bad-whitespace
 
     theta = np.arccos(t)
-    BS_tw, _, _ = BSgate(theta, phi, cutoff=trunc)
+    BS_tw = beamsplitter_tw(theta, phi, cutoff=trunc)
 
-    # TODO: Transpose needed because of different conventions in SF and The Walrus. Remove when The Walrus is updated.
+    # Transpose needed because of different conventions in SF and The Walrus.
```
Comment (antalszava) [inline strawberryfields/backends/fockbackend/ops.py:349]: Just curious: so this means that the difference in convention remains with the update in The Walrus, correct? (Slightly connected to similar changes proposed in #378)

**Comment by josh146** [comment]
I've updated the `requirements.txt` file to install the latest Walrus dev version. Should we merge this in now? A couple of PRs are currently blocked due to master not having latest Walrus dev compatibility, including #378 and #407.

**Pull request merged**